ext_mongodb.php

```diff
@@ -667,7 +667,31 @@ function bsonUnserialize(array $data) : void;
 
 final class Binary implements Type, \Serializable
 {
-	use DenySerialization;
+	static private function checkArray(array $state)
+	{
+		if (
+			!array_key_exists( 'data', $state ) || !is_string( $state['data'] )
+			||
+			!array_key_exists( 'type', $state ) || !is_int( $state['type'] )
```

Comment:
I find "isset" always being a bit wonky. So I prefer doing the array key test first.